src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -3,133 +3,538 @@
+	params: CommonParams,
+	processFeature: (userFeature: DevContainerFeature) => Promise<FeatureSet | undefined>,
+	worklist: FNode[],
+	acc: FNode[],
+	legacyAcc: FNode[]): Promise<DependencyGraph> {
+	const { output } = params;
 
-    if (config.overrideFeatureInstallOrder) {
-        return computeOverrideInstallationOrder(config, features);
-    }
-    else {
-        return computeInstallationOrder(features);
-    }
+	while (worklist.length > 0) {
+		const current = worklist.shift()!;
+
+		output.write(`Resolving dependencies for '${current.userFeatureId}'...`, LogLevel.Info);
+
+		const processedFeature = await processFeature(current);
+		if (!processedFeature) {
+			throw new Error(`ERR: Feature '${current.userFeatureId}' in dependency graph could not be processed.  You may not have permission to access this Feature.  Please report this to the Feature author.`);
+		}
+
+		// Set the processed FeatureSet object onto Node.
+		current.featureSet = processedFeature;
+
+		// If the Feature doesn't support dependencies,  add it to the
+		// legacy accumulator but do not attempt to process its dependencies (since it cannot have any!)
+		if (!featureSupportsDependencies(processedFeature)) {
+			legacyAcc.push(current);
+			continue;
+		}
+
+		// If the current Feature is already in the accumulator, skip it.
+		// This stops cycles but doesn't report them.  
+		// Cycles/inconsistencies are thrown as errors in the next stage (rounds).
+		if (acc.some(f => equals(f, current))) {
+			continue;
+		}
+
+		const type = processedFeature.sourceInformation.type;
+		let metadata: Feature | undefined;
+		// Switch on the source type of the provided Feature.
+		// Retrieving the metadata for the Feature (the contents of 'devcontainer-feature.json')
+		switch (type) {
+			case 'oci':
+				metadata = await getOCIFeatureMetadata(params, current);
+				break;
+
+			case 'direct-tarball':
+				throw new Error(`Feature type 'direct-tarball' will be supported but is unimplemented.`);
+
+			case 'file-path':
+				const filePath = (current.featureSet.sourceInformation as FilePathSourceInformation).resolvedFilePath;
+				const metadataFilePath = path.join(filePath, DEVCONTAINER_FEATURE_FILE_NAME);
+				if (!isLocalFile(filePath)) {
+					throw new Error(`Metadata file '${metadataFilePath}' cannot be read for Feature '${current.userFeatureId}'.`);
+				}
+				const serialized = (await readLocalFile(metadataFilePath)).toString();
+				if (serialized) {
+					metadata = jsonc.parse(serialized) as Feature;
+				}
+				break;
+
+			default:
+				throw new Error(`Feature dependencies are only supported in Features published (2) to an OCI registry, (2) as direct HTTPS tarball, or (3) as a local Feature.  Got type: '${processedFeature.sourceInformation.type}'.`);
+		}
+
+		// Resolve dependencies given the current Feature's metadata.
+		if (metadata) {
+			current.featureSet.features[0] = {
+				...current.featureSet.features[0],
+				...metadata,
```

Comment:
Good catch, working on this.

Reply:
Maybe instead of using `userFeatureId` directly (e.g., in `compareTo`), you could have a function `normalizedFeatureId(feature: FeatureSet)` that returns the current feature id? I guess that only matters for OCI features, in all other cases this function would return `userFeatureId`?

Reply:
Ah I am just seeing this now.  I did something similar where it considers all combination of legacyIds and the current Id.